modules/unspents/src/dimensions.ts

```diff
@@ -1,287 +1,151 @@
-  return Math.ceil(getInputComponentsWeight(components) / 4);
-}
+import { VirtualSizes } from './virtualSizes';
+export { VirtualSizes };
 
-// Constants for signed TX input and output vsizes.
-// See https://bitcoincore.org/en/segwit_wallet_dev/#transaction-serialization for full description
-// FIXME(BG-9233): use weight units instead
-export const VirtualSizes = Object.freeze({
-  // FIXME(BG-7873): add support for signature grinding
-
-  // Size of a P2PKH input with (un)compressed key
-  /** @deprecated */
-  txP2pkhInputSizeCompressedKey: 148,
-  /** @deprecated */
-  txP2pkhInputSizeUncompressedKey: 180,
-
-  // Input sizes
-  txP2shInputSize: getVirtualInputSizeFromComponents(inputComponentsP2sh),
-  txP2shP2wshInputSize: getVirtualInputSizeFromComponents(inputComponentsP2shP2wsh),
-  txP2wshInputSize: getVirtualInputSizeFromComponents(inputComponentsP2wsh),
-  txP2trKeypathInputSize: getVirtualInputSizeFromComponents(inputComponentsP2trKeySpend),
-  txP2shP2pkInputSize: getVirtualInputSizeFromComponents(inputComponentsP2shP2pk),
-  txP2trScriptPathLevel1InputSize: getVirtualInputSizeFromComponents(inputComponentsP2trScriptSpendLevel1),
-  txP2trScriptPathLevel2InputSize: getVirtualInputSizeFromComponents(inputComponentsP2trScriptSpendLevel2),
-
-  //
-  // Output sizes
-  //
-
-  // The size is calculated as
-  //
-  //    scriptLength + compactSize(scriptLength) + txOutputAmountSize
-  //
-  // Since compactSize(scriptLength) is 1 for all scripts considered here, we can simplify this to
-  //
-  //    scriptLength + 9
-  //
-
-  // Size of single output amount
-  txOutputAmountSize: 8,
-
-  // https://github.com/bitcoinjs/bitcoinjs-lib/blob/v4.0.2/src/templates/scripthash/output.js#L9
-  txP2shOutputSize: 32,
-  txP2shP2wshOutputSize: 32,
-  // https://github.com/bitcoinjs/bitcoinjs-lib/blob/v4.0.2/src/templates/witnessscripthash/output.js#L9
-  txP2wshOutputSize: 43,
-  // OP_1 OP_PUSH32 <schnorr_public_key>
-  txP2trOutputSize: 43,
-  // https://github.com/bitcoinjs/bitcoinjs-lib/blob/v4.0.2/src/templates/pubkeyhash/output.js#L9
-  txP2pkhOutputSize: 34,
-  // https://github.com/bitcoinjs/bitcoinjs-lib/blob/v4.0.2/src/templates/witnesspubkeyhash/output.js#L9
-  txP2wpkhOutputSize: 31,
-
-  /** @deprecated - use txP2pkhOutputSize instead */
-  txOutputSize: 34,
-
-  //
-  // General tx size constants
-  //
-
-  txOverheadSize: 10,
-  // Segwit adds one byte each for marker and flag to the witness section.
-  // Thus, the vsize is only increased by one.
-  txSegOverheadVSize: 11,
-});
-
-export interface IOutputDimensions {
-  count: number;
-  size: number;
+/**
+ * Apply `f` to all properties of `d`
+ */
+function mapDimensions(
+  d: Partial<Dimensions>,
+  f: <T extends keyof Dimensions>(key: T, v: Dimensions[T] | undefined) => unknown
+): Dimensions {
+  return new Dimensions(
+    Object.fromEntries(Object.entries(d).map(([key, value]) => [key, f(key as keyof Dimensions, value)]))
```

Comment:
I'm a bit surprised the `as keyof Dimensions` is necessary here, I would've thought that'd be implied from the `Parital<Dimensions>` type.

Edit: Just checked this and `key` is `string` type so the cast is necessary.